### PR TITLE
[PLA-9117] Remove the experimental fields.

### PR DESCRIPTION
**Pull request by anoto-moniz**
# Citrine Python PR

## Description 
It's been hard-coded in the backend to always be false with an empty
list of reasons. It's not accessed in many places, but we do need to
deprecate it where we do. But mostly, this just rips it out and updates
the tests, including adding tests that a deprecation warning is raised.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/__version__.py
Comment (kroenlein) [inline src/citrine/__version__.py:1]:
```suggestion
__version__ = '1.25.0'
```
I think philosophically this is more of a minor bump given that anyone who had been using this functionality would need to change their behavior at this point.  Not required for approval.

File: src/citrine/_rest/ai_resource_metadata.py
Comment (kroenlein) [inline src/citrine/_rest/ai_resource_metadata.py:40]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

File: src/citrine/_rest/ai_resource_metadata.py
Comment (kroenlein) [inline src/citrine/_rest/ai_resource_metadata.py:46]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

File: src/citrine/informatics/executions/design_execution.py
Comment (kroenlein) [inline src/citrine/informatics/executions/design_execution.py:105]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

File: src/citrine/informatics/executions/design_execution.py
Comment (kroenlein) [inline src/citrine/informatics/executions/design_execution.py:111]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

File: src/citrine/informatics/executions/predictor_evaluation_execution.py
Comment (kroenlein) [inline src/citrine/informatics/executions/predictor_evaluation_execution.py:90]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

File: src/citrine/informatics/executions/predictor_evaluation_execution.py
Comment (kroenlein) [inline src/citrine/informatics/executions/predictor_evaluation_execution.py:96]:
```suggestion
    @deprecated(deprecated_in="1.25.0", removed_in="2.0.0")
```
As above

**Review by kroenlein — DISMISSED** [review]
Question about version choice, but generally looks solid.

File: src/citrine/__version__.py
Comment (anoto-moniz) [reply src/citrine/__version__.py:1]: I think that's a fair point.

Comment (anoto-moniz) [reply src/citrine/__version__.py]: The one thing I dislike about GitHub's PR system is that no matter what, if you push a change, it invalidates approvals. It makes sense to a certain degree, of course, but I do wish we could say "Approve even if they make changes".

**Review by kroenlein — APPROVED** [review]
LGTM

**Pull request merged**
